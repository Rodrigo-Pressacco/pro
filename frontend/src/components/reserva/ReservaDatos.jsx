import React, { useEffect, useState } from "react";
import "../../styles/reserva/ReservaDatos.css";


const ReservaDatos = (props) => {
    
    let localStorageToken = null
    const [usuarioSetter, setUsuarioSetter] = useState(false)
    const [usuario, setUsuario] = useState({
        nombre: "",
        apellido: "",
        email: ""
    })

    let covid = document.getElementsByName('covid').value;

    useEffect(() => {
        if (!usuarioSetter) {
            setUsuario(JSON.parse(localStorage.getItem("myKeyForPersona")))
            localStorageToken = JSON.parse(localStorage.getItem("myKeyForToken"))
            setUsuarioSetter(true)
        }
    });

    const handleClick = (data)=>{
        if(data.target.id==="covid-si"){
            props.covid(1);
        }else{
            props.covid(0);
        }
        
    }

    const handleChangeCiudad = e => {
        props.ciudadUsuario(e.target.value);
        
    }

    const handleChangeMensaje = e => {
        props.mensajeDueño(e.target.value);
        }
    


    return (
        <form action="">
            <div className="form-top">
                <div className="form-datos">
                    <label >Nombre:</label>
                    <input type="text" name="nombre" id="nombre" disabled="disabled" value={usuario.nombre} />
                </div>
                <div className="form-datos">
                    <label >Apellido:</label>
                    <input type="text" name="apellido" id="apellido" disabled="disabled" value={usuario.apellido} />
                </div>
                <div className="form-datos">
                    <label >Correo Electronico:</label>
                    <input type="text" name="email" id="email" disabled="disabled" value={usuario.email} />
                </div>
                <div className="form-datos">
                    <label >* Ciudad:</label>
                    <input type="text" name="ciudad" id="ciudad" onChange={handleChangeCiudad}/>
                </div>
            </div>
            <div className="form-botton">
                <div className="form-mensaje">
                    <label htmlFor="msjDueño">Mensaje para el dueño:</label>
                    <textarea name="msjDueño" id="msjDueño" onChange={handleChangeMensaje} ></textarea>

                </div>
                <div className="obligatorio">
                    <div className="form-covid">
                        <input type="radio" name="covid" id="covid-si" onClick={handleClick} />
                        <label >Si</label>
                        <input type="radio" name="covid" id="covid-no" onClick={handleClick} />
                        <label >No</label>                        
                        <label >* ¿Estas vacunado para el Covid-19?</label>
                    </div>
                    <p>Los campos con * son obligatorios</p>
                </div>
            </div>
        </form>
    )
}
//<input type="text" name="" id="" />
export default ReservaDatos;