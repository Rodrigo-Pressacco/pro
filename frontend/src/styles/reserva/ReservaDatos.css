

.form-top{
    display: flex;
    flex-wrap: wrap;
    justify-content: space-evenly;
    align-items: center;
}

.form-botton{
    display: flex;
    flex-direction: column;
}

.form-datos{
    width: 50%;
    display: flex;
    flex-direction: column;
}

.form-datos label{
    height: 20px;
    margin: 20px 0 10px 20px;
    font-style: normal;
    font-weight: 500;
    font-size: 16px;
    line-height: 14px;
    color: #383B58;
}

.obligatorio{
    display: flex;
    align-items: center;
    justify-content: space-between;
}

.obligatorio p{
    margin-right: 25px;
        color: #808080cc;
        font-size: 12px;
}

.form-datos input{
    margin: 0 20px 0 20px;
    width: 90%;
    height: 40px;
    border: none;
    background: #DFE4EA;
    opacity: 0.4;
    box-shadow: 0px 1px 5px rgba(0, 0, 0, 0.15);
    border-radius: 5px;
    padding-left: 15px;
    
    font-style: normal;
    font-weight: bold;
    font-size: 16px;
    line-height: 19px;
}

.form-mensaje{    
    width: 100%;
    display: flex;
    flex-direction: column;
}

.form-mensaje label{
    height: 20px;
    margin: 20px 0 10px 20px;
    font-style: normal;
    font-weight: 500;
    font-size: 16px;
    line-height: 14px;
    color: #383B58;
}

.form-mensaje textarea{
    margin: 0 20px 0 20px;
    border: none;
    background: #DFE4EA;
    opacity: 0.4;
    box-shadow: 0px 1px 5px rgba(0, 0, 0, 0.15);
    border-radius: 5px;
    height: 40px;
    padding-top: 5px;
    padding-left: 15px;
    min-height: 40px;
    max-width: 95%;
    
    font-style: normal;
    font-weight: bold;
    font-size: 16px;
    line-height: 19px;
}

.form-covid{
    display: flex;
    flex-direction: row;
    padding: 20px;
    align-items: center;
}

.form-covid label{
    margin-left: 10px;    
    font-style: normal;
    font-weight: 500;
    font-size: 16px;
    line-height: 14px;
    color: #383B58;
}

.form-covid input{
    width: 20px;
    height: 20px;
}

#ciudad,
#msjDue√±o{
    background-color: #fff;
    box-shadow: 0px 1px 5px rgba(0, 0, 0, 25%);
    color: #545776;
    opacity: 1;

}

#covid-si,
#covid-no{
    margin-left: 4px;
}

@media (max-width: 485px){

    .form-datos{
        width: 100%;
    }
    
    .form-datos input {
        margin: 0 10px 0 10px;
    }

    .form-mensaje textarea{
        max-width: 88%;
    }

    .obligatorio p{
        margin-right: 0px;
        display: none;
    }
}