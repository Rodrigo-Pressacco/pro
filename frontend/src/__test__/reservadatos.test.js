import React from "react";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import ReservaDatos from '../components/reserva/ReservaDatos'

import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { createSerializer } from 'enzyme-to-json';
Enzyme.configure({ adapter: new Adapter() });
expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));

describe('<ReservaDatos />', () => {
    let reservaDatos
    let useEffect = jest.fn(); 
    

    beforeEach(() => {
        reservaDatos = shallow(<ReservaDatos useEffect={useEffect} />)
    });

    it("Deberia mostrar <ReservaDatos/> correctamente", () => {
        expect(reservaDatos).toMatchSnapshot();
    });

    it("Deberia retornar 8 etiquetas label", () => {
        let node = reservaDatos.find("label")
        expect(node.length).toEqual(8)
    });

    it("Deberia retornar 6 etiquetas input", () => {
        let node = reservaDatos.find("input")
        expect(node.length).toEqual(6)
    });

    it("Deberia retornar 1 etiqueta textarea", () => {
        let node = reservaDatos.find("textarea")
        expect(node.length).toEqual(1)
    });

    it("Deberia retornar 1 etiqueta p", () => {
        let node = reservaDatos.find("p")
        expect(node.length).toEqual(1)
    })

})