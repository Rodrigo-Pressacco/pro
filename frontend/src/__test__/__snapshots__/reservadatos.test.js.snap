// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReservaDatos /> Deberia mostrar <ReservaDatos/> correctamente 1`] = `
<form
  action=""
>
  <div
    className="form-top"
  >
    <div
      className="form-datos"
    >
      <label>
        Nombre:
      </label>
      <input
        disabled="disabled"
        id="nombre"
        name="nombre"
        type="text"
        value=""
      />
    </div>
    <div
      className="form-datos"
    >
      <label>
        Apellido:
      </label>
      <input
        disabled="disabled"
        id="apellido"
        name="apellido"
        type="text"
        value=""
      />
    </div>
    <div
      className="form-datos"
    >
      <label>
        Correo Electronico:
      </label>
      <input
        disabled="disabled"
        id="email"
        name="email"
        type="text"
        value=""
      />
    </div>
    <div
      className="form-datos"
    >
      <label>
        * Ciudad:
      </label>
      <input
        id="ciudad"
        name="ciudad"
        type="text"
      />
    </div>
  </div>
  <div
    className="form-botton"
  >
    <div
      className="form-mensaje"
    >
      <label
        htmlFor="msjDueño"
      >
        Mensaje para el dueño:
      </label>
      <textarea
        id="msjDueño"
        name="msjDueño"
      />
    </div>
    <div
      className="obligatorio"
    >
      <div
        className="form-covid"
      >
        <input
          id="covid-si"
          name="covid"
          type="radio"
        />
        <label>
          Si
        </label>
        <input
          id="covid-no"
          name="covid"
          type="radio"
        />
        <label>
          No
        </label>
        <label>
          * ¿Estas vacunado para el Covid-19?
        </label>
      </div>
      <p>
        Los campos con * son obligatorios
      </p>
    </div>
  </div>
</form>
`;
